examples/apps/color_finder.py

```diff
@@ -0,0 +1,475 @@
+
+        text = cozmo.annotate.ImageText('Looking for {}'.format(self.color_to_find), color='white')
+        text.render(d, (0, 0, image.width, image.height))
+
+    def on_cube_tap(self, evt, **kwargs):
+        ''' The blinking white cube switches the Tkviewer between normal mode and pixel mode.
+            The other illuminated cube toggles self.color_to_find.
+        '''    
+        if kwargs['obj'].object_id == self.color_selector_cube.object_id:
+            self.toggle_color_to_find()
+        elif kwargs['obj'].object_id == self.grid_cube.object_id:
+            self.robot.world.image_annotator.annotation_enabled = not self.robot.world.image_annotator.annotation_enabled
+
+    def toggle_color_to_find(self):
+        ''' Sets self.color_to_find to the next color in self.possible_colors_to_find, 
+            wrapping back around to the initial color if we have reached the end of the list.
+        '''    
+        self.color_to_find_index+=1
+        if self.color_to_find_index == len(self.possible_colors_to_find):
+            self.color_to_find_index = 0
+        self.color_to_find = self.possible_colors_to_find[self.color_to_find_index]
+        self.color_selector_cube.set_lights(map_color_to_light[self.color_to_find])
+
+    def update_pixel_matrix(self,downsized_image):
+        ''' Uses the getpixel method from the Pillow library to get the RGB values
+            of all the pixels in the low-res approximation of Cozmo's camera view.
+        '''
+        for i in range(self.grid_width):
+            for j in range(self.grid_height):
+                r,g,b = downsized_image.getpixel((i,j))
+                self.pixel_matrix[i][j] = self.approximate_color_of_pixel(r,g,b)
+        self.fill_in_gaps_of_pixel_matrix()
+
+    def approximate_color_of_pixel(self,r,g,b):
+        ''' Returns the approximate color of this pixel from color_dict.'''
+        min_distance = sys.maxsize
+        closest_color = ''
+        for color_name, color_values in color_dict.items():
+            d = color_distance((r,g,b),color_values)
+            if d < min_distance:
+                min_distance = d
+                closest_color = color_name
+        return closest_color
+
+    def fill_in_gaps_of_pixel_matrix(self):
+        ''' Fills in squares in self.pixel_matrix that meet the condition
+            in self.pixel_is_surrounded_by_color
+        '''
+        for i in range(1,self.grid_width-1):
+            for j in range(1,self.grid_height-1):
+                color = self.pixel_is_surrounded_by_color(i,j)
+                if color:
+                    self.pixel_matrix[i][j]=color
+
+    def pixel_is_surrounded_by_color(self,i,j):
+        ''' If a pixel is surrounded by three pixels of the same color, returns that color. Otherwise, returns None.'''
+        if self.pixel_matrix[i-1][j]==self.pixel_matrix[i][j-1] and self.pixel_matrix[i-1][j]==self.pixel_matrix[i+1][j]:
```

Comment:
Should either handle edge cases where e.g. i==0, or document it (as it is I had to look at the calling function to see that it wasn't calling it for 0 or width-1 values)
I think this would read more cleanly if you pulled the pixels out, e.g:
```python
left_pixel = self.pixel_matrix[i-1][j]
up_pixel = self.pixel_matrix[i][j-1]
```